src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -946,7 +946,7 @@ public void onRefreshSettings(Settings settings) {
         @Override
         public void run() {
             // we check before if a refresh is needed, if not, we reschedule, otherwise, we fork, refresh, and then reschedule
-            if (!engine().refreshNeeded()) {
+            if (!engineSafe().refreshNeeded()) {
```

Comment:
Why do we have a separate engineSafe() method?  Can't we just move that to engine() and remove engineSafe()?  It makes me feel like engine() is unsafe :)